setuptools/command/egg_info.py

```diff
@@ -575,6 +575,12 @@ def add_defaults(self):
             self.filelist.extend(rcfiles)
         elif os.path.exists(self.manifest):
             self.read_manifest()
+        if os.path.exists("setup.py"):
+            """If there exists a setup.py, it should be included in the sdist irrespective
```

Comment:
maybe would make sense to check that it's not already included, what happens if we double include stuff? E.g. via manifest and here. If it's ignored is ok as it is. We check for its existence but how and when would it not be available? Isn't setuptools pre requisites to exist?

Reply:
Ideally you'd be able to just specify `setup.cfg` and `setuptools` as a build backend, and you could get away with no `setup.py` at all.

My preference would be a world where the *common case* is that there's no `setup.py` (though that's basically impossible at the moment).